components/encoder/fake/encoder.go

```diff
@@ -149,21 +146,20 @@ func (e *fakeEncoder) Properties(ctx context.Context, extra map[string]interface
 	}, nil
 }
 
+// Close safely shuts down the fake encoder.
+func (e *fakeEncoder) Close(ctx context.Context) error {
+	e.mu.Lock()
+	defer e.mu.Unlock()
```

Comment:
No need to lock the mutex: it protects the position, updateRate, and similar things, but we're not using any of those.